src/swsssdk/dbconnector.py

```diff
@@ -27,91 +93,136 @@ def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
                 logger.warning(msg)
                 sonic_db_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config', 'database_config.json')
             with open(sonic_db_file_path, "r") as read_file:
-                SonicDBConfig._sonic_db_config = json.load(read_file)
+                # The database_config.json is loaded into the '' index, which referes to the local namespace.
+                SonicDBConfig._sonic_db_config[''] = json.load(read_file)
         except (OSError, IOError):
             msg = "Could not open sonic database config file '{}'".format(sonic_db_file_path)
             logger.exception(msg)
             raise RuntimeError(msg)
         SonicDBConfig._sonic_db_config_init = True
 
     @staticmethod
-    def db_name_validation(db_name):
+    def namespace_validation(namespace):
+        # Check the namespace is valid.
+        if namespace is None:
```

Comment:
why not treating None as global namespace?

Reply:
As discussed, changed the default value of namespace parameter as None. If the namespace is coming as None in any of the API's -- it is converted internally to the '' string referring to local namespace.

Here we check for namespace is None for a scenario where user calls this SonicDBConfig.namespace_validation() API explicitly and passes None. We error out in that case.